doc/tutorials/electrodes/electrodes_part1.ipynb

```diff
@@ -0,0 +1,500 @@
+    "charges = {\"Cation\": q[0], \"Anion\": -q[0]  }\n",
+    "\n",
+    "p1=system.part.add(pos=[box_l_x/4.0, box_l_y/2.0, box_l_z/2.0], q=charges[\"Cation\"])\n",
+    "print(f\"Cation placed at position: {p1.pos}\")\n",
+    "p2=system.part.add(pos=[3.0*box_l_x/4.0, box_l_y/2.0, box_l_z/2.0], q=charges[\"Anion\"])\n",
+    "print(f\"Anion placed at position: {p2.pos}\")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "solution2": "hidden",
+    "solution2_first": true
+   },
+   "source": [
+    "### Task:\n",
+    "Set up electrostatics interactions between the ion pairs using P3M and ICC $\\star$.  \n",
+    "#### Hints:\n",
+    "- First instantiate a P3M object (using\n",
+    "  [*espressomd.electrostatics.P3M(...)*](https://espressomd.github.io/doc/espressomd.html#espressomd.electrostatics.P3M))\n",
+    "  as the 3D Coulomb solver for the system.\n",
+    "- Since, we are working on a 2D+h partially periodic system, we use ELC in\n",
+    "  combination with P3M. For this purpose, instantiate ELC (using\n",
+    "  [*espressomd.electrostatics.ELC(...)*](https://espressomd.github.io/doc/espressomd.html#espressomd.electrostatics.ELC))\n",
+    "  with p3m as actor, which calls necessary initialization routines.  \n",
+    "- Now we account for dielectric interface by adding ICC $^\\star$.\n",
+    "  To this end, first create an ICC object using\n",
+    "  [*espressomd.electrostatic_extensions.ICC(...)*](https://espressomd.github.io/doc/espressomd.html#espressomd.electrostatic_extensions.ICC).\n",
+    "- Setting up ICC $^\\star$ requires specifying a fixed number of ICC particles\n",
+    "  and their initial charges.\n",
+    "  Since the ICC particles are normal ESPResSo particles, they need to be added\n",
+    "  at the interfaces using\n",
+    "  [*system.part.add(..)*](https://espressomd.github.io/doc/espressomd.html?#module-espressomd.particle_data).\n",
+    "\n",
+    "*Note* - Refer to section\n",
+    "[**Dielectric interfaces with the ICC algorithm**](https://espressomd.github.io/doc/electrostatics.html#dielectric-interfaces-with-the-icc-star-algorithm)\n",
+    "in the **ESPResSo** documentation for the basics of an ICC $^\\star$ setup.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "solution2": "hidden",
+    "solution2_first": true
+   },
+   "outputs": [],
+   "source": [
+    "p3m = espressomd.electrostatics.P3M(\n",
+    "            prefactor=BJERRUM_LENGTH,\n",
+    "            accuracy=ACCURACY,\n",
+    "            check_neutrality = False,\n",
+    "            mesh = [100,100,150],\n",
+    "            cao = 5\n",
+    "        )\n",
+    "\n",
+    "elc = espressomd.electrostatics.ELC(actor=p3m, gap_size=ELC_GAP, maxPWerror=CHECK_ACCURACY)\n",
+    "\n",
+    "# Set the ICC line density and calculate the number of\n",
+    "# ICC particles according to the box size\n",
+    "nicc = 100  # linear density\n",
```

Comment:
this variable needs to be defined outside of the task, the hints must include a point that helps with the particle number calculation

Reply:
also nicc is a very misleading name for a density